tests/pytests/unit/modules/test_schedule.py

```diff
@@ -67,24 +68,25 @@ def test_purge(sock_dir, job1):
 
     changes = {"job1": "removed", "job2": "removed", "job3": "removed"}
 
-    schedule_config_file = schedule._get_schedule_config_file()
-    with patch.dict(
-        schedule.__opts__, {"schedule": {"job1": "salt"}, "sock_dir": sock_dir}
-    ):
-        with patch("salt.utils.files.fopen", mock_open(read_data="")) as fopen_mock:
-            with patch.object(
-                schedule, "list_", MagicMock(return_value=_schedule_data)
-            ):
-                ret = schedule.purge(offline=True)
-                assert any([True for item in comm if item in ret["comment"]])
-                assert ret["changes"] == changes
-                assert ret["result"]
+    with patch("os.makedirs", MagicMock(return_value=True)):
+        schedule_config_file = schedule._get_schedule_config_file()
+        with patch.dict(
+            schedule.__opts__, {"schedule": {"job1": "salt"}, "sock_dir": sock_dir}
+        ):
+            with patch("salt.utils.files.fopen", mock_open(read_data="")) as fopen_mock:
+                with patch.object(
+                    schedule, "list_", MagicMock(return_value=_schedule_data)
+                ):
+                    ret = schedule.purge(offline=True)
```

Comment:
**suggestion (non-blocking)** same patch-then-single-with story here, though it'd be something like `with patch_mkdirs, patch_opts, patch_fopen as fopen_mock, patch_schedule_list:`